src/components/Filters/StickyOptions/StickyOptions.js

```diff
@@ -0,0 +1,97 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import AggregationItem from '../AggregationItem/AggregationItem';
+
+const mapOfOptions = (options) => {
+  const result = options.reduce((map, opt) => {
+    map[opt.key] = opt;
+    return map;
+  }, {});
+  return result;
+};
+
+const zeroCounts = (cache) => {
+  const result = {};
+  Object.keys(cache).forEach((key) => {
+    result[key] = {
+      ...cache[key],
+      // eslint-disable-next-line camelcase
+      doc_count: 0,
+    };
+  });
+
+  return result;
+};
+
+const StickyOptions = ({
+  fieldName,
+  options,
+  selections = [],
+  onMissingItem = (item) => ({
+    key: item,
+    // eslint-disable-next-line camelcase
+    doc_count: 0,
+  }),
+}) => {
+  const [trackedSelections, setTrackedSelections] = useState(
+    selections.slice(),
+  );
+  const [cache, setCache] = useState(mapOfOptions(options));
+
+  const updateValues = () => {
+    // Zero out the counts in the cache
+    const zeroed = zeroCounts(cache);
+
+    // Update the cache with the new values
+    // and zero out the rest
+    const updatedCache = Object.assign(zeroed, mapOfOptions(options));
+
+    // always additive (the options are "sticky")
+    const toBeTrackedSelections = [...trackedSelections].slice();
+
+    selections.forEach((selection) => {
+      // Add any new selections
+      if (toBeTrackedSelections.indexOf(selection) === -1) {
+        toBeTrackedSelections.push(selection);
+      }
+
+      // Add missing cache options
+      if (!(selection in updatedCache)) {
+        updatedCache[selection] = onMissingItem(selection);
```

Comment:
It doesn't look like StickyOptions gets onMissingItem passed in anywhere, and we should just inline this here. 

```
        updatedCache[selection] = {
          key: selection,
          // eslint-disable-next-line camelcase
          doc_count: 0,
        };
```